webapi/src/main/scala/org/knora/webapi/routing/Authenticator.scala

```diff
@@ -462,39 +600,117 @@ object Authenticator {
-                    // didn't found one, so I will try to get it from the triple store
-                    val userProfileV1Future = for {
-                        maybeUserProfileV1 <- (responderManager ? UserProfileByEmailGetV1(email, UserProfileType.FULL)).mapTo[Option[UserProfileV1]]
-                        userProfileV1 = maybeUserProfileV1 match {
-                            case Some(up) => up
-                            case None => {
-                                log.debug(s"getUserProfileByEmail - supplied email not found - throwing exception")
-                                throw BadCredentialsException(s"$BAD_CRED_USER_NOT_FOUND")
-                            }
-                        }
-                        _ = CacheUtil.put(AUTHENTICATION_CACHE_NAME, email, userProfileV1)
-                        _ = log.debug(s"getUserProfileByEmail - from triplestore: $userProfileV1")
-                    } yield userProfileV1
-
-                    // TODO: return the future here instead of using Await.
-					Await.result(userProfileV1Future, Duration(3, SECONDS))
-            }
+            val userProfileV1Future = for {
+                maybeUserProfileV1 <- (responderManager ? UserProfileByEmailGetV1(email, UserProfileTypeV1.FULL)).mapTo[Option[UserProfileV1]]
+                userProfileV1 = maybeUserProfileV1 match {
+                    case Some(up) => up
+                    case None => {
+                        log.debug(s"getUserProfileV1ByEmail - supplied email not found - throwing exception")
+                        throw BadCredentialsException(s"$BAD_CRED_USER_NOT_FOUND")
+                    }
+                }
+                _ = log.debug(s"getUserProfileV1ByEmail - userProfileV1: $userProfileV1")
+            } yield userProfileV1
+
+            // TODO: return the future here instead of using Await.
+            Await.result(userProfileV1Future, Duration(3, SECONDS))
+        } else {
+            throw BadCredentialsException(BAD_CRED_EMAIL_NOT_SUPPLIED)
+        }
+    }
+}
+
+object JWTHelper {
+
+    import Authenticator.AUTHENTICATION_INVALIDATION_CACHE_NAME
+
+    val algorithm = Algorithm.HS256
+    val requiredHeaders: Set[HeaderField] = Set[HeaderField](Typ)
+    val requiredClaims: Set[ClaimField] = Set[ClaimField](Iss, Sub, Aud, Iat, Exp, Jti)
+
+    val log = Logger(LoggerFactory.getLogger(this.getClass))
+
+    /**
+      * Create a JWT.
+      *
+      * @param userIri   the user IRI that will be encoded into the token.
+      * @param secretKey the secret key used for encoding.
+      * @param longevity the token's longevity in days.
+      * @return a [[String]] containg the JWT.
+      */
+    def createToken(userIri: IRI, secretKey: String, longevity: Long): String = {
+
+        val headers = Seq[HeaderValue](Typ("JWT"), Alg(algorithm))
+
+        val now: Long = System.currentTimeMillis() / 1000l
+        val nowPlusLongevity: Long = now + longevity * 60 * 60 * 24
```

Comment:
Please add a comment saying how much time this is.

Reply:
it's user configurable in `application.conf`:

```
app {
     ...
     // the JSON Web Token longevity in days.
     jwt-longevity = 30
     ...
}